src/v/cluster/topic_table.cc

```diff
@@ -538,55 +492,57 @@ topic_table::wait_for_changes(ss::abort_source& as) {
 
 std::vector<model::topic_namespace> topic_table::all_topics() const {
     return transform_topics(
-      [](const topic_configuration_assignment& td) { return td.cfg.tp_ns; });
+      [](const topic_metadata& tp) { return tp.get_configuration().tp_ns; });
 }
 
-std::optional<model::topic_metadata>
+// TODO: copy
+std::optional<topic_metadata>
 topic_table::get_topic_metadata(model::topic_namespace_view tp) const {
     if (auto it = _topics.find(tp); it != _topics.end()) {
-        return it->second.configuration.get_metadata();
+        return it->second;
+    }
+    return {};
+}
+std::optional<std::reference_wrapper<const topic_metadata>>
+topic_table::get_topic_metadata_ref(model::topic_namespace_view tp) const {
+    if (auto it = _topics.find(tp); it != _topics.end()) {
+        return it->second;
```

Comment:
I find the lifetime issues a little bit mind-mending to thing about. I satisfied myself that this function is at least correct I think: we return a reference (well a `std::optional<std::reference_wrapper<const topic_metadata>>` but this amounts to `const topic_metadata&` from a lifetime perspective, right?), but because this is a view which itself contains reference, we are OK as long as the underlying topic_namespace lives long enough, right?

The topic table is kept up to date through the metadata dissemination service, is that right? So if a topic goes away, this view may be invalid, so we should not hold it across a possible suspension point?

Reply:
this is perfect explanation. We can not hold that `refrence_wrapper` between the scheduling points, hence i created a `...ref` method. On the other hand there are many places where we need to access topic partitions for some operations like filtering of validation, those operations are synchronous. Having a method returning reference will prevent copies of a large partitions set for each of those operations.

Reply:
Great, @mmaslankaprv !

It would be nice to have a little summary of this in the comment for the function :).

Reply:
and perhaps in the commit message please :)